backend/api/Services/MissionSchedulingService.cs

```diff
@@ -452,33 +424,9 @@ private async Task StartMissionRun(MissionRun queuedMissionRun)
             return ongoingMissions;
         }
 
-        private async Task<bool> TheSystemIsAvailableToRunAMission(string robotId, string missionRunId)
+        private bool TheSystemIsAvailableToRunAMission(Robot robot, MissionRun missionRun)
         {
-            bool ongoingMission = await OngoingMission(robotId);
-
-            if (ongoingMission)
-            {
-                logger.LogInformation("Mission run {MissionRunId} was not started as there is already an ongoing mission", missionRunId);
-                return false;
-            }
-
-            var robot = await robotService.ReadById(robotId, readOnly: true);
-            if (robot is null)
-            {
-                string errorMessage = $"Robot with ID: {robotId} was not found in the database";
-                logger.LogError("{Message}", errorMessage);
-                throw new RobotNotFoundException(errorMessage);
-            }
-
-            var missionRun = await missionRunService.ReadById(missionRunId, readOnly: true);
-            if (missionRun is null)
-            {
-                string errorMessage = $"Mission run with Id {missionRunId} was not found in the database";
-                logger.LogError("{Message}", errorMessage);
-                throw new MissionRunNotFoundException(errorMessage);
-            }
-
-            if (robot.MissionQueueFrozen && missionRun.MissionRunType != MissionRunType.Emergency)
+            if (robot.MissionQueueFrozen && !(missionRun.IsEmergencyMission() || missionRun.IsReturnHomeMission()))
```

Comment:
how is the ordering on this check? Would it go ((frozen && not emergency) or returnhomemission)?

Reply:
frozen && not (emergency or returnhome)